### ISSUE TITLE
Properly resize window with SDL2

### ISSUE THREAD

**Issue by stephanemagnenat** [Globulation2/glob2#41]
Now we are using SDL2, but the window resize is done by closing it and opening a new one, which is not ideal. A proper resize using `SDL_SetWindowSize` should be implemented.

**Comment by Quipyowert2** [comment]
I've tried to implement resizing. Without OpenGL mode checked (opengl in a VM is rather slow), it works okay in my virtual machine ( Linux guest; Windows 10 host) but when compiled natively for Windows (using MSYS2 Mingw 64-bit shell) I ran into bug libsdl-org/SDL#1059. The workarounds mentioned in that bug report include using:
* SDL_SetWindowHitTest
* SetTimer/KillTimer
* SDL_AddEventWatch and calling redrawing function there.
* Moving rendering into separate thread.

**Comment by stephanemagnenat** [comment]
Do you think that the [workaround using `SDL_SetEventFilter`](https://github.com/ppy/osu-framework/commit/c938e6c9094cfec5d1dd30cbd27b9735e311d363) could work for us?

**Comment by Quipyowert2** [comment]
Yes, I think a solution using SDL_SetEventFilter/SDL_AddEventWatch might work, but I had trouble with it. My older implementation ([render-thread](https://github.com/Quipyowert2/glob2/tree/render-thread)) uses way too many lambdas and is probably the wrong way to go about it, and has assertion failures all the time. Then I have a handle-resize branch that enables resize events which is the base for my `eventfilter` and `redraw` branches which use basically the same idea of giving GraphicContext the objects to call the redraw function on. My redraw branch has a problem: the globule moves for only 1 tick whenever the mouse cursor moves when resizing on Windows and stops moving when I stop moving the cursor but continue holding mouse button down.

**Comment by stephanemagnenat** [comment]
Should we first merge a minimal in which we accept that the main thread is blocked while resizing on some platforms?
